pkg/beholder/message.go

```diff
@@ -0,0 +1,257 @@
+package beholder
+
+import (
+	"fmt"
+
+	"github.com/go-playground/validator/v10"
+	"go.opentelemetry.io/otel/attribute"
+	otellog "go.opentelemetry.io/otel/log"
+)
+
+type Message struct {
+	Attrs Attributes
+	Body  []byte
+}
+
+type Metadata struct {
+	//	REQUIRED FIELDS
+	// Schema Registry URI to fetch schema
+	BeholderDataSchema string `validate:"required,uri"`
+
+	// OPTIONAL FIELDS
+	// The version of the CL node.
+	NodeVersion string
+	// mTLS public key for the node operator. This is used as an identity key but with the added benefit of being able to provide signatures.
+	NodeCsaKey string
+	// Signature from CSA private key.
+	NodeCsaSignature string
+	DonID            string
+	// The RDD network name the CL node is operating with.
+	NetworkName          []string
+	WorkflowID           string
+	WorkflowName         string
+	WorkflowOwnerAddress string
+	// Hash of the workflow spec.
+	WorkflowSpecID string
+	// The unique execution of a workflow.
+	WorkflowExecutionID string
+	// The address for the contract.
+	CapabilityContractAddress string
+	CapabilityID              string
+	CapabilityVersion         string
+	CapabilityName            string
+	NetworkChainID            string
+}
+
+func (m Metadata) Attributes() Attributes {
+	return Attributes{
+		"node_version":                m.NodeVersion,
+		"node_csa_key":                m.NodeCsaKey,
+		"node_csa_signature":          m.NodeCsaSignature,
+		"don_id":                      m.DonID,
+		"network_name":                m.NetworkName,
+		"workflow_id":                 m.WorkflowID,
+		"workflow_name":               m.WorkflowName,
+		"workflow_owner_address":      m.WorkflowOwnerAddress,
+		"workflow_spec_id":            m.WorkflowSpecID,
+		"workflow_execution_id":       m.WorkflowExecutionID,
+		"beholder_data_schema":        m.BeholderDataSchema,
+		"capability_contract_address": m.CapabilityContractAddress,
+		"capability_id":               m.CapabilityID,
+		"capability_version":          m.CapabilityVersion,
+		"capability_name":             m.CapabilityName,
+		"network_chain_id":            m.NetworkChainID,
+	}
+}
+
+type Attributes map[string]any
```

Comment:
I think I understand why you ended up creating `Attributes`. Just to explore a bit - is this over-engineering? I don't see how this makes anything simpler for the consumer. Is there another way to accomplish this? 

cc @jmank88

Reply:
Using an alias would disallow methods but also drop any required conversions to/from the basic map type :shrug: 
```suggestion
type Attributes = map[string]any
```

Reply:
> I don't see how this makes anything simpler for the consumer.

If all attributes are constructed next to the `Emit` call then Attributes and sugar methods will not be of any use.

These are use cases I had in mind originally, see here https://github.com/smartcontractkit/chainlink-common/pull/694/files#diff-0ebe30d99f28fddcfef8e717720321077f917a7fe18aa1207d49c3b989812e79R16

After the code in `message.go` has evolved I can see that having it as a `map[string]any` alias makes more sense. 

Removing sugar methods, see here https://github.com/smartcontractkit/chainlink-common/commit/4e80921db29d6f144890789656570af35a17f528

Reply:
https://github.com/smartcontractkit/chainlink-common/commit/4e80921db29d6f144890789656570af35a17f528